### ISSUE TITLE
Koha-Ceepos-maksurajapintaa käytettäessä Hyväksy-painike epäaktiiviseksi

### ISSUE THREAD

**Issue by pirliis** [KohaSuomi/Koha#351]
### Kehitysehdotukseni

Kun asiakkaan maksu on lähetetty maksurajapintaa käyttäen Kohasta Ceeposiin, olisi erittäin hyvä, että Maksa-näytöllä näkyvä Hyväksy-painike (jolla voi poistaa maksun Kohasta manuaalisesti) menisi "epäaktiiviseksi", jottei virkailija pysty maksun lähetyksen jälkeen enää poistamaan maksua manuaalisesti. 

OUTIssa tulee esille tapauksia, jossa virkailija on ensin lähettänyt maksun Kohasta Ceeposiin ja tämän jälkeen poistanut maksun Kohasta Hyväksy-painikkeella. Tämän jälkeen virkailija on veloittanut lähetetyn maksun asiakkaalta Ceeposissa, jolloin maksurajapinta on myös lähettänyt Kohaan tiedon, että maksu on maksettu. Asiakkaan maksut menevät tässä tapauksessa miinukselle maksetun summan verran ja asiakkaan Lainaus- ja Tiedot-sivuilla on huomautus: "Asiakkaan tilillä on hyvitys suuruudeltaan x,xx.

 

### Jotain muuta?

Hyväksy-painike epäaktiiviseksi sen jälkeen, kun maksu on lähetetty Kohasta Ceeposiin käyttäen Ceeposmaksu-painiketta.
![image](https://user-images.githubusercontent.com/98814860/210527927-6468c0c9-dd25-4306-8696-e56d01a43c93.png)
Tieto asiakkaan Maksut-sivun Tapahtumat-välilehdellä:
![image](https://user-images.githubusercontent.com/98814860/210526198-bed4ce74-32bc-4d15-94fb-6bcf38941411.png)
Tieto asiakkaan Lainaus- ja Tiedot-sivuilla:
![image](https://user-images.githubusercontent.com/98814860/210526808-18325a43-0caf-4d23-891a-75f1835b5a57.png)

**Comment by PiaKusmin** [comment]
Lapissa samaa ongelmaa. Toivotaan Oulun ehdottamaa korjausta.

**Comment by pirliis** [comment]
Olisiko tätä kehitysehdotusta mahdollista toteuttaa? OUTIssa on jatkuvasti asiakkaita, joilla on miinusmerkkisiä maksusaldoja. Henkilökuntaa on ohjeistettu ja muistutettu, mitä pitää tehdä ja mitä ei saa tehdä, kun maksut lähetetään Kohasta Ceeposiin, mutta aina siellä on joku, joka tekee niin kuin ei saisi tehdä. :(

**Comment by AnneliO** [comment]
Tämä ehdotus ei ole ehtinyt vielä käymään asiantuntijaryhmässä, koska olemme käsitelleet vielä Redminestä vanhoja häntiä. Ne kuitenkin saatiin nyt viime kokouksessa käsiteltyä ja päästään vihdoin näihin GitHubissa oleviin kehitysehdotuksiin.

**Comment by AnneliO** [comment]
Koha-Suomen asiantuntijaryhmä 5.2.2024: Piilotetaan Hyväksy-nappi Ceeposta käyttävistä yksiköistä.

**Comment by AnneliO** [comment]
Liitännäisen Readmessä taitaa olla vanhempi versio rimpsusta. Uusin versio on [IntranetUserJS-kirjastossa](https://koha-suomi.fi/dokumentaatio/intranetuserjs/#ceepos-maksu--napin-lis%C3%A4ys). Pitäneekin muistaa päivittää kumpaankin paikkaan sitten muutetettu versio.

**Comment by atarnus** [comment]
Joo. Tehdään nykyisen lausekkeen alkuun tuo muutos. Alla nyt esimerkki miten tämä on Outi-testissä. Outin lausekkeessa on näköjään myös muutettu let-muuttujat var-muuttujiksi, ja on logattu aika paljon asiaa. Katsotaan nyt mikä on sitten se final form :).
Testattavissa nyt Outi-testillä:

```
/// ALKU ///

/// Ceepos-maksu -napin lisäys ///
/// Napin taustavärinä ja reunavärinä keltainen? #ffc32b ///
$(document).ready(function() {
  var ceeposBranches = ['OUPK','OUR','OUHA','OUMA','OUUL','OUOS','OUH','OUKI','OUKS','OUKA','OUPE','UTPK','IIPK','PUPK','VAPK','LUPK'];
  if (ceeposBranches.includes($("#logged-in-info-full .logged-in-branch-code").text())) {
    $("#payfine .action, #payindivfine .action").find("input").hide().after('<input type="button" id="CeeposMaksu" class="btn btn-primary" style="margin:0px 10px;"  value="Ceepos-maksu"  onclick="setCeeposPayment($(this))"/>');
```

Muuttunut rivi erikseen vielä tässä:
`    $("#payfine .action, #payindivfine .action").find("input").hide().after('<input type="button" id="CeeposMaksu" class="btn btn-primary" style="margin:0px 10px;"  value="Ceepos-maksu"  onclick="setCeeposPayment($(this))"/>');`

**Comment by pknuuti** [comment]
Onko tätä mahdollisuus testata muillakin testeillä vai testaako Outi yksin?

**Comment by pknuuti** [comment]
Vaaran testillä todettu, että jos kirjastotunnus on Ceepos-listassa, vain Ceepos-maksu -painike ja Peruuta-painike ovat näkyvissä maksuja maksettaessa. Jos kirjastotunnusta ei ole Ceepos-listauksessa, maksutilanteessa näkyy vain Hyväksy tai Peruuta-painike, ei Ceeposta ollenkaan.

**Comment by LeenaKinnunen** [comment]
Lapin testillä toimii samoin kuin Kyytissä, eli toimii.

**Comment by pirliis** [comment]
Myös OUTIssa painikkeet nyt samoin kuin Vaarassa eli ihan toimivalta ratkaisulta näyttää. :)

**Comment by atarnus** [comment]
Koodimuutokset tulivat siis riveille 6-16, eli if-lauseke (1. ja viim. rivi tässä) ja nappirivi (6. tässä):

```
  if (window.location.pathname == '/cgi-bin/koha/members/paycollect.pl' && !window.location.search.includes("WRITEOFF")) {
    var ceeposBranches = ['OUPK','OUR','OUHA','OUMA','OUUL','OUOS','OUH','OUKI','OUKS','OUKA','OUPE','UTPK','IIPK','PUPK','VAPK','LUPK'];
    if (ceeposBranches.includes($("#logged-in-info-full .logged-in-branch-code").text())) {
      console.log("ceeposbranch löytyi");
      console.log($("#logged-in-info-full .logged-in-branch-code").text()); 
      $("#payfine .action, #payindivfine .action").find("input").hide().after('<input type="button" id="CeeposMaksu" class="btn btn-primary" style="margin:0px 10px;" value="Ceepos-maksu"  onclick="setCeeposPayment($(this))"/>');
      if(localStorage.getItem('ceeposOffice')) {
        $('#payment_type').val(localStorage.getItem('ceeposOffice'));
      }
    }
  }
```
